dan_layer/consensus/src/hotstuff/on_inbound_message.rs

```diff
@@ -35,11 +35,11 @@ pub struct OnInboundMessage<TConsensusSpec: ConsensusSpec> {
     leader_strategy: TConsensusSpec::LeaderStrategy,
     pacemaker: PaceMakerHandle,
     vote_signing_service: TConsensusSpec::SignatureService,
-    rx_hotstuff_message: mpsc::Receiver<(TConsensusSpec::Addr, HotstuffMessage)>,
+    pub rx_hotstuff_message: mpsc::Receiver<(TConsensusSpec::Addr, HotstuffMessage)>,
     tx_outbound_message: mpsc::Sender<(TConsensusSpec::Addr, HotstuffMessage)>,
     tx_msg_ready: mpsc::UnboundedSender<(TConsensusSpec::Addr, HotstuffMessage)>,
-    rx_new_transactions: mpsc::Receiver<TransactionId>,
-    message_buffer: MessageBuffer<TConsensusSpec::Addr>,
+    pub rx_new_transactions: mpsc::Receiver<TransactionId>,
```

Comment:
nit: structs with a mix of pub and private fields is code smell (couldn't find a good write up but that is my two cents ;) - also https://users.rust-lang.org/t/structs-with-public-fields-are-brittle/989)
I see why you had to do that though due to multiple mutable references, but we should clean this up in future.

Reply:
yeah, I tried to do it another way, but as you said, multiple mutable references was the problem